### PR TITLE
Prefer pattern matching over key access

### PR DESCRIPTION
**Pull request by barthez**
Recently I read [article](https://medium.com/learn-elixir/speed-up-data-access-in-elixir-842617030514) about the `Access` module and ways of accessing `Map` values in Elixir. It turned out that accessing values by e.g. `job.name` is slower than pattern matching. 

I thought I could improve your great library with this small change. What do you think @c-rack?

I was also thinking about improving specifications and typing in affected files. Is it worth it?

### CODE REVIEW

**Comment by maennchen** [comment]
Thank you very much @barthez. I’m going to have a closer look on monday. It looks very good on first glance.

**Comment by c-rack** [comment]
Thanks!

**Comment by c-rack** [comment]
@barthez Any improvement to this project is appreciated.

File: lib/quantum/job_broadcaster.ex [hunk 1]
```diff
@@ -52,35 +47,32 @@ defmodule Quantum.JobBroadcaster do
...
-    {:noreply, [], put_in(state[:jobs][job.name], job)}
+    {:noreply, [], %{state | jobs: Map.put(jobs, job_name, job)}}
   end
 
   def handle_cast({:delete, name}, %{jobs: jobs} = state) do
     Logger.debug(fn ->
       "[#{inspect(Node.self())}][#{__MODULE__}] Deleting job #{inspect(name)}"
     end)
 
-    cond do
```
Comment (maennchen) [inline lib/quantum/job_broadcaster.ex]: The job is not deleted from state anymore when it is inactive.

**Comment by maennchen** [comment]
@c-rack I would recommend waiting for my review for a PR like that. This PR effectively creates unpredictable behaviour given the right circumstances. I'm going to create a new PR to fix it.

File: lib/quantum/job_broadcaster.ex (hunk 1 above)
Comment (barthez) [reply lib/quantum/job_broadcaster.ex]: And before my the change it wasn't removed either.

Maybe tests should be updated to check expected behavior.

Comment (maennchen) [reply lib/quantum/job_broadcaster.ex]: Oh yeah, you're right about that. I'm going to add a test.

**Pull request merged**